common/src/main/java/com/microsoft/identity/common/internal/ui/webview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -88,8 +87,21 @@ public EmbeddedWebViewAuthorizationStrategy(@NonNull Activity activity) {
     @Override
     public void completeAuthorization(int requestCode, int resultCode, Intent data) {
         if (requestCode == BROWSER_FLOW) {
-            final AuthorizationResult result = mOAuth2Strategy.getAuthorizationResultFactory().createAuthorizationResult(resultCode, data, mAuthorizationRequest);
-            mAuthorizationResultFuture.setResult(result);
+            if (mOAuth2Strategy != null && mAuthorizationResultFuture != null) {
+                final AuthorizationResult result = mOAuth2Strategy
+                        .getAuthorizationResultFactory()
+                        .createAuthorizationResult(
+                                resultCode,
+                                data,
+                                mAuthorizationRequest
+                        );
+                mAuthorizationResultFuture.setResult(result);
+            } else {
+                Logger.warn(TAG, "SDK Cancel triggering before request is sent out. " +
```

Comment:
If we get a log from customer, would this line be enough for investigation (as in to know what exactly happened)?

Reply:
Good point, added if `oAuth2Strategy` and `mAuthorizationResultFuture ` are null, i.e. both should be null and there is nothing we need to do as there is no future waiting to be ended, in any case only one of them is null then we have another problem which are not aware of.